department-of-veterans-affairs/caseflow
# Add panel/clerk/per curium options to AddCavcRemandView judge selector

Implements https://vajira.max.gov/browse/CASEFLOW-984

### Description
We now support "Clerk", "Panel", and "Per Curium" as judge "names" in addition to selecting a judge's name.

Note that you must restart your Rails server locally or it will throw a validation error when you try to save this. I don't think this will be an issue in prod since clients shouldn't get this before the servers do.

### Acceptance Criteria
- Given a CAVC Litigation Support user is completing the CAVC entry form
  - when they select the the "What is the CAVC judge's name?" dropdown 
    -  [x] Then the following additional options are available: 
       - Clerk
       - Panel
       - Per Curium

### Testing Plan
1. Pull up the form to enter a new CAVC remand¹
2. Ensure that "Clerk", "Panel", and "Per Curium" are options at the top of the judge list
3. Fill out the form normally with one of those options selected and submit
4. Ensure that selection is displayed as the judge under the CAVC Remand section when the form is submitted

¹ Accessing the form:

```ruby
RequestStore[:current_user] = User.system_user

original_appeal = FactoryBot.create(:appeal, 
                           :dispatched, 
                           associated_judge: JudgeTeam.first.judge, 
                           associated_attorney: JudgeTeam.first.attorneys.first);
FactoryBot.create_list(:request_issue, 2,                                                                                  
            :rating,                                                                                            
            :with_rating_decision_issue,                                                                           
            decision_review: original_appeal,                                                                               
            veteran_participant_id: Veteran.first.participant_id,                                                        
            contested_issue_description: "issue_description",                                                              
            notes: "issue_notes");

original_appeal.uuid
```

Then navigate to /queue/appeals/UUID and click "Add CAVC Remand".


### User Facing Changes

N.B. - In testing this, I observed that the judge selection input takes the full width of the screen. This behavior predates my PR, though.

Before:
![Screen Shot 2021-02-23 at 11 55 56 AM](https://user-images.githubusercontent.com/314663/108878355-2fc3b580-75ce-11eb-9ef9-1052ce5c12d1.png)

After:
![Screen Shot 2021-02-23 at 11 38 40 AM](https://user-images.githubusercontent.com/314663/108878393-38b48700-75ce-11eb-9aa4-60429c4ac5db.png)

Demonstrating that it displays correctly if submitted:

![Screen Shot 2021-02-23 at 11 02 16 AM](https://user-images.githubusercontent.com/314663/108878585-6699cb80-75ce-11eb-94bb-8921ca95c5fc.png)
